### PR TITLE
Update issue template

### PR DESCRIPTION
**Pull request by mcbarton**
This PR updates the issue template for the repo to complement the PR which adds a pull request template here https://github.com/compiler-research/xeus-cpp/pull/118

### CODE REVIEW

**Comment by mcbarton** [comment]
@anutosh491 @JohanMabille can you please review this PR and the other one linked?

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -1,14 +1,29 @@
-* xeus-cookiecutter version:
+# Checklist
+Before submitting an issue please confirm you have checked the follow
+- [ ] This issue is not a duplicate of one already open
+- [ ] The title of the issue is clear
```
Comment (JohanMabille) [inline .github/ISSUE_TEMPLATE.md:4]: I would add a "This issue is not described in the documentation" checkbox here.

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
+
+# System information
+* xeus-cpp version:
 * Operating System:
+* CPU architechture:
```
Comment (JohanMabille) [inline .github/ISSUE_TEMPLATE.md:9]: Typo: architechture => architecture
I would add examples of architectures in parenthesis, as beginners may not understand what this means

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
+
+## Description of issue
+Please tick the boxes which apply to your issue
+- [ ] Bug in xeus-cpp
+- [ ] Request for new feature to be added
+- [ ] Issue with documentation
+- [ ] Other (please state below)
```
Comment (JohanMabille) [inline .github/ISSUE_TEMPLATE.md:16]: This should be handled by GH label, when we do triage. Some of the previous sections may not apply to all the possibilities here. Also the other sections make me think that this template is for bug reports.

**Comment by mcbarton** [comment]
@JohanMabille can you take another look at this PR? I fixed all the issues you raised except switching to Github labels. With regards to the Github labels issue I plan to add a workflow which automatically tags each new issue a 'Needs Review' label (needs creating if you could do this for me since I don't have the permissions to do this). It will then be upto the person who reviews the issue to add the correct labels based on the description of the issue.

**Comment by vgvassilev** [comment]
Can you try again to create a label? I added you to the repo.

**Comment by mcbarton** [comment]
> Can you try again to create a label? I added you to the repo.

Thank you I am now able to add a label :)

**Comment by mcbarton** [comment]
@JohanMabille I have now addressed all your issues with this PR and its ready for another review.

**Comment by JohanMabille** [comment]
Can I suggest to use the label "Needs triage" instead of "Needs review" for new issues? It is more widespread in OSS repos (at least on GH). otherwise, LGTM.

**Comment by mcbarton** [comment]
> Can I suggest to use the label "Needs triage" instead of "Needs review" for new issues? It is more widespread in OSS repos (at least on GH). otherwise, LGTM.

@JohanMabille done.

**Comment by JohanMabille** [comment]
Thanks!

**Pull request merged**